lib/configuration/configuration.js

```diff
@@ -112,15 +112,29 @@ class Configuration {
     }
   }
 
-  static instanceWithContext (context) {
+  static fetchConfigFile (context) {
     let github = context.github
     let repo = context.repo()
 
-    return github.repos.getContent({
-      owner: repo.owner,
-      repo: repo.repo,
-      path: Configuration.FILE_NAME
-    }).then(res => {
+    if (context.event === 'pull_request') {
+      return github.repos.getContent({
+        owner: repo.owner,
+        repo: repo.repo,
+        path: Configuration.FILE_NAME,
+        ref: context.payload.pull_request.head.ref
+      })
```

Comment:
What happens here when there is no Configuration changes or additions in the PR? I think in that scenario it still needs to retrieve from the repo "master"

Reply:
OK sounds good. If that's how the `github.repos.getContent` we are gtg.